### PR TITLE
Add a route-like link to the modal

### PR DESCRIPTION
**Pull request by nburka**
Maybe there's a better way to do this? Trying to do it JS-only.

This opens a modal with the URL `/#icon/filled/name`

Not sure if there's some better way. I tried following some tutorials like this one https://github.com/vercel/next.js/tree/canary/examples/with-route-as-modal and ran into problems with that approach. If we do go that way, we have to make sure that works with the statically published version of the site.

This works pretty well though.

### CODE REVIEW

**Comment by nburka** [comment]
Fixes https://github.com/resolvetosavelives/healthicons/issues/80

File: src/pages/index.tsx [hunk 1]
```diff
@@ -48,33 +52,63 @@ export default function Home({ categories }: HomeProps) {
...
+  const ogImagePath = modalIcon
+    ? `icons/png/${modalIcon.iconType}/${modalIcon.icon.path}/${modalIcon.icon.fileName}@2x.png`
+    : 'og_image.png';
+
+  const ogImageWidth = modalIcon ? '96' : '1280';
+  const ogImageHeight = modalIcon ? '96' : '640';
+
   return (
     <div className="container">
       <Head>
```
Comment (nburka) [inline src/pages/index.tsx:87]: this whole section should probably be its own component. Either now, or soon.

**Comment by sgarrity** [comment]
Tested this out and I think there's an edge case that doesn't work as intended:

1. Open the site
2. Click on an icon (we'll call it icon B)
3. See that the URL is updated (perfect!), copy that URL for icon B.
4. Open a new tab and paste in the new URL (works as intended - you see the modal for icon B)
5. Now, close the modal or open another one, and stay on the site. Paste in the URL we copied, and hit Enter. For me, it's not navigating to the modal for icon B - it's just staying as is.

I think in Step 5 it should be opening the modal for icon B, but isn't.

**Comment by sgarrity** [comment]
Also, do we need the `#` in the URL? I'm not sure we do.

**Comment by nburka** [comment]
> Also, do we need the `#` in the URL? I'm not sure we do.

I'm not sure either. The example on NextJS uses their router dir/filename structure. I tried that and ran into problems trying to load all of the `prop.categories`. I could use some help on this.

**Comment by dburka** [comment]
Is this ready to go @nburka @sgarrity ?

**Comment by nburka** [comment]
this is superseded by https://github.com/resolvetosavelives/healthicons/pull/119

**Pull request closed without merging**